### ISSUE TITLE
bug(prefer-object-from-entries): false negative when flattening array of objects

### ISSUE THREAD

**Issue by privatenumber** [sindresorhus/eslint-plugin-unicorn#1631]
The `prefer-object-from-entries` rule reports that I should use `Object.fromEntries()` when I use `Array#reduce` to flatten an array of objects:

```
  27:38  error    Prefer `Object.fromEntries()` over `Array#reduce()`  unicorn/prefer-object-from-entries
```

It should not suggest that `Object.fromEntries()` be used here because it doesn't do the same thing:

```js
const arrayOfObjects = [
	{ a: 1 },
	{ b: 2 },
	{ c: 3 }
];

const flattened = arrayOfObjects.reduce((flattened, next) => Object.assign(flattened, next), {});
// => { a: 1, b: 2, c: 3 }
```

**Comment by fisker** [comment]
```js
Object.fromEntries(arrayOfObjects.flatMap(object => Object.entries(object)))
```

**Comment by intellix** [comment]
Also getting this myself. Isn't an Object.assign spread better?

```
Object.assign(...arrayOfObjects); // {a: 1, b: 2, c: 3}
```

**Comment by privatenumber** [comment]
If that's what the rule is actually recommending I do, I would say that's quite a leap from the message it gives: "Prefer `Object.fromEntries()` over `Array#reduce()`"

Perhaps what's actionable here is updating the rule to provide more specific suggestions. For example, if `flatMap` and `Object.entries` should also be used, that should be in the suggestion message too.
